Towny/src/main/java/com/palmergames/bukkit/towny/db/TownyFlatFileSource.java

```diff
@@ -530,7 +531,10 @@ public boolean loadResident(Resident resident) {
 				
 				line = keys.get("friends");
 				if (line != null) {
-					List<Resident> friends = TownyAPI.getInstance().getResidents(line.split(","));
+					final String[] split = line.split(",");
+					final UUID[] friendUUIDs = toUUIDArray(split);
+
+					List<Resident> friends = friendUUIDs.length > 0 ? api.getResidents(friendUUIDs) : api.getResidents(split);
```

Comment:
So the thinking here is that if the DB is still holding resident names for  a friend list that friendUUIDs will have a length of 0?